graphistry/PlotterBase.py

```diff
@@ -2458,6 +2495,112 @@ def cypher(self, query: str, params: Dict[str, Any] = {}) -> Plottable:
             )\
             .nodes(nodes)\
             .edges(edges)
+    
+    
+    def spanner_gql_to_g(self: Plottable, query: str) -> Plottable:
+        """
+        Submit GQL query to google spanner graph database and return Plottable with nodes and edges populated  
+        
+        GQL must be a path query with a syntax similar to the following, it's recommended to return the path with
+        SAFE_TO_JSON(p), TO_JSON() can also be used, but not recommend. LIMIT is optional, but for large graphs with millions
+        of edges or more, it's best to filter either in the query or use LIMIT so as not to exhaust GPU memory.  
+
+        query=f'''GRAPH my_graph
+        MATCH p = (a)-[b]->(c) LIMIT 100000 return SAFE_TO_JSON(p) as path'''
+
+        :param query: GQL query string 
+        :type query: Str
+
+        :returns: Plottable with the results of GQL query as a graph
+        :rtype: Plottable
+
+        **Example: calling spanner_gql_to_g
+                ::
+
+                    import graphistry
+
+                    # credentials_file is optional, all others are required
+                    SPANNER_CONF = { "project_id":  PROJECT_ID,                 
+                                     "instance_id": INSTANCE_ID, 
+                                     "database_id": DATABASE_ID, 
+                                     "credentials_file": CREDENTIALS_FILE }
+
+                    graphistry.register(..., spanner_config=SPANNER_CONF)
+
+                    query=f'''GRAPH my_graph
+                    MATCH p = (a)-[b]->(c) LIMIT 100000 return SAFE_TO_JSON(p) as path'''
+
+                    g = graphistry.spanner_gql_to_g(query)
+
+                    g.plot()
+     
+        """
+        from .pygraphistry import PyGraphistry
+
+        res = copy.copy(self)
+        
+        if res._spannergraph is None: 
+            spanner_config = PyGraphistry._config["spanner"]
+            if spanner_config is not None: 
+                logger.debug(f"Spanner Config: {spanner_config}")
+            else: 
+                logger.warning('PyGraphistry._config["spanner"] is None')
+        
+            res = res.spanner_init(PyGraphistry._config["spanner"])  # type: ignore[attr-defined]
+
+        return res._spannergraph.gql_to_graph(query)
```

Comment:
this seems a bit off, b/c  instead of `gql_to_graph` working with the settings of `self: Plottable`, it'll be using the `g` at time of `register()`

i think it should be a call like `res._spannergraph.gql_to_graph(self, query)` or `res._spannergraph.gql_to_graph(res, query)`